001-intro.md

```diff
@@ -100,96 +100,358 @@ to real life cases involving actual DMs.**
+astronomical domain, e.g. catalogues of astronomical sources. These are the models
+the IVOA sanctions in standards.
+
+However, different missions, or archives, or applications will most certainly
+have specific additional features that are not captured by such common, standardized
+data models.
+
+One may express such extensions and their instances in such a way that:
+  * data providers may easily annotate specialized instances, including the additional information, in a
+  standardized, interoperable fashion.
+  * clients of the common, standard data models may still find instances of these parent models in files
+  serializing specialized instanced.
+  * a model annotation that is valid according to a specialized data model
+  is also valid according to the parent model.
+
+This use case can be formalized in terms of inheritance and polymorphism.
+
+Inheritance allows
+models to specialize types defined in other data models. Polymorphism is
+the common object-oriented design concept that says that the value of a
+property may be an instance of a *subtype* of the declared type.
+
+Typed languages such as Java support a casting operation, which provides
+more information to the interpreter about the type it may expect a
+certain instance to be.
+
+A client must be able to identify
+the information about a standard type, even if the serialization includes
+instances of its subtypes. Similarly, a client should have enough information to *cast* an instance
+from the declared type to the actual subtype.
+
+Growing complexity: simple, advanced, and guru clients {#sec:clients}
+-----------------------------------------------------
+
+According to the use cases depicted above, we can classify clients
+in terms of how they parse the VOTable in order to harvest its content.
+Of course, in the real word such distinction is somewhat fuzzy, but this
+section tries and describe the different levels of usage of this
+specification.
+
+This classification is useful because it shows how implementations can be based
+on different assumptions. Some clients can focus on few hard-coded elements,
+while other clients can dynamically address complex tasks.
+
+### Simple clients ###
+
+We say that a client is *simple* if:
+
+  * it does not parse the VO-DML description file
+
+  * it assumes the a priori knowledge of one or more data models
+
+  * it discovers information by looking for a set of predefined
+    vodml-refs in the VOTable
+
+In other terms, a simple client has knowledge of the data model it is
+sensitive to, and simply discovers information useful to its own use
+cases by traversing the `VODML` element.
+
+Examples of such clients are the DAL service clients that allow users to
+discover and fetch datasets. They may just inspect the response of a
+service and present the user with a subset of its metadata. They do not
+*reason* on the content, and they are not interested in the structure of
+the serialized objects.

```

Comment:
@lmichel 
Your comment:
> Section 2.4.1: You are suggesting the DAL clients are not model aware. This not always true. For instance, SSA DAL responses rely on SDM.

I don't think that is what I am suggesting, otherwise this text would not be in this specification to begin with. What I am defining is a class of clients that have a-priori knowledge of a model, or a subset of a model, which is what you are describing for SSA clients and I am extending to all *simple* DAL clients.

Maybe there was a mix up of the section numbers and we are commenting on different bits of the text?

> The point is that there no logical link between the model and the DAL response. All entities dealing with SSA are supposed to refer to the same model but there is no way ensure that this is true

I am not sure I understand this part of the comment and how it relates to simple clients as defined in this section.